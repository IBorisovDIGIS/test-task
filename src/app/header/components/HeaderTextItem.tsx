import React from 'react';
import {
  Grid, makeStyles, Typography,
} from '@material-ui/core';
import { gray, mainText } from '../../../constants/colors';

const useStyles = makeStyles({
  label: {
    flexGrow: 1,
    fontSize: 13,
    color: gray,
  },
  title: {
    flexGrow: 1,
    fontSize: 19,
    color: mainText,

  },
});

interface IHeader {
    label: string;
    title: string;
}

const HeaderTextItem: React.FC<IHeader> = ({ label, title }: IHeader) => {
  const classes = useStyles();

  return (
    <Grid item>
      <Typography noWrap className={classes.label}>
        {label}
      </Typography>
      <Typography noWrap className={classes.title}>
        {title}
      </Typography>
    </Grid>
  );
};

export default HeaderTextItem;
