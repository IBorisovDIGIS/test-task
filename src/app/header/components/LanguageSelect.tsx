import React, { useState } from 'react';
import { MenuItem, Select } from '@material-ui/core';
import { useTranslation } from 'react-i18next';
import { languages } from '../../../constants';

const LanguageSelect: React.FC = () => {
  const { i18n } = useTranslation();
  const [language, setLanguage] = useState(languages[0]);

  const handleSetLanguage = async (event: React.ChangeEvent<{ value: unknown }>) => {
    const lang = event.target.value as string;
    await i18n.changeLanguage(lang);
    setLanguage(lang);
  };

  return (
    <Select onChange={handleSetLanguage} value={language}>
      {languages.map((lang) => <MenuItem value={lang} key={lang}>{lang.toUpperCase()}</MenuItem>)}
    </Select>
  );
};

export default LanguageSelect;
