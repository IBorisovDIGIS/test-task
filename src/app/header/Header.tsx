import React from 'react';
import clsx from 'clsx';
import {
  AppBar, Grid, makeStyles, Toolbar,
} from '@material-ui/core';
import { useTranslation } from 'react-i18next';
import HeaderTextItem from './components/HeaderTextItem';
import { white } from '../../constants/colors';
import LanguageSelect from './components/LanguageSelect';
import { drawerWidth } from '../../constants';

const useStyles = makeStyles((theme) => ({
  toolbar: {
    paddingRight: 24,
    background: white,
    shadow: 'none',
  },
  toolbarIcon: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-end',
    padding: '0 8px',
  },
  appBar: {
    marginLeft: theme.spacing(9),
    width: `calc(100% - ${theme.spacing(9)}px)`,
    zIndex: theme.zIndex.drawer + 1,
    transition: theme.transitions.create(['width', 'margin'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
  },
  appBarShift: {
    marginLeft: drawerWidth,
    width: `calc(100% - ${drawerWidth}px)`,
    transition: theme.transitions.create(['width', 'margin'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  menuButton: {
    marginRight: 36,
  },
  menuButtonHidden: {
    display: 'none',
  },
  title: {
    flexGrow: 1,
  },

}));

interface IHeader {
    open: boolean;
    balance: number,
    next_payout: number,
    currency: string;
}

const Header: React.FC<IHeader> = ({
  open, balance, next_payout, currency,
}: IHeader) => {
  const classes = useStyles();
  const { t } = useTranslation();
  const displayCurrency = currency === 'usd' ? '$' : '';

  return (
    <AppBar position="absolute" className={clsx(classes.appBar, open && classes.appBarShift)}>
      <Toolbar className={classes.toolbar}>
        <Grid container spacing={2}>
          <HeaderTextItem label={t('balance')} title={`${balance.toLocaleString()} ${displayCurrency}`} />
          <HeaderTextItem label={t('payout')} title={`${next_payout.toLocaleString()} ${displayCurrency}`} />
        </Grid>
        <LanguageSelect />
      </Toolbar>
    </AppBar>
  );
};

export default Header;
