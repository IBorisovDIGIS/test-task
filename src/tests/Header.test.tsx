import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {AppBar, Grid, Toolbar, Typography} from '@material-ui/core';
import toJson from 'enzyme-to-json';
import Header from '../app/header/Header';
import HeaderTextItem from "../app/header/components/HeaderTextItem";
import LanguageSelect from "../app/header/components/LanguageSelect";

configure({ adapter: new Adapter() });

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: string) => key })
}));

describe('Header should render correctly', () => {
  const headerProps = { open: false, balance: 0, next_payout: 1, currency: 'usd' };

  it('Should renders correctly', () => {
    const container = shallow(<Header {...headerProps} />);

    expect(toJson(container)).toMatchSnapshot();
  });

  it('Should render elements correctly', () => {
    expect(AppBar).toBeDefined();
    expect(Toolbar).toBeDefined();
    expect(Grid).toBeDefined();
    expect(HeaderTextItem).toBeDefined();
    expect(LanguageSelect).toBeDefined();

  });

  it('Should render balance correctly', () => {
    const container = shallow(<Header {...headerProps} />);

    const balance = container.find(HeaderTextItem).first().shallow()
    const title = balance.find(Typography).first().text();
    const description = balance.find(Typography).last().text();
    expect(title).toEqual('balance')
    expect(description).toContain(headerProps.balance)
  });

  it('Should render payout correctly', () => {
    const container = shallow(<Header {...headerProps} />);

    const payout = container.find(HeaderTextItem).last().shallow()
    const title = payout.find(Typography).first().text();
    const description = payout.find(Typography).last().text();
    expect(title).toEqual('payout')

    expect(description).toContain(headerProps.next_payout)
  });

});
