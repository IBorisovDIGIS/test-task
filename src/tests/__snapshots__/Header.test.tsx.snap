// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly Should renders correctly 1`] = `
<WithStyles(ForwardRef(AppBar))
  className="makeStyles-appBar-3"
  position="absolute"
>
  <WithStyles(ForwardRef(Toolbar))
    className="makeStyles-toolbar-1"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={2}
    >
      <HeaderTextItem
        label="balance"
        title="0 $"
      />
      <HeaderTextItem
        label="payout"
        title="1 $"
      />
    </WithStyles(ForwardRef(Grid))>
    <LanguageSelect />
  </WithStyles(ForwardRef(Toolbar))>
</WithStyles(ForwardRef(AppBar))>
`;
